packages/vscode-dependencies-validation/src/autofix/depsIssues.ts

```diff
@@ -0,0 +1,59 @@
+import type { Uri } from "vscode";
+import {
+  VscodeFileEventConfig,
+  VscodeUriFile,
+  VscodeWorkspace,
+} from "../vscodeTypes";
+import {
+  ENABLE_AUTOFIX,
+  getAutoFixDelay,
+  isAutoFixEnabled,
+} from "./configuration";
+import { addPackageJsonFileWatcher as addPackageJsonFileWatcher } from "./packageJsonFileWatcher";
+import { addUnsupportedFilesWatcher } from "./unsupportedFilesWatcher";
+import { findAndFixDepsIssues } from "../util";
+import { clearDiagnostics } from "../util";
+
+export function activateDepsIssuesAutoFix(
+  vscodeConfig: VscodeFileEventConfig & VscodeUriFile
+): void {
+  fixWorkspaceDepsIssues(vscodeConfig);
+  addPackageJsonFileWatcher(vscodeConfig);
+  addUnsupportedFilesWatcher(vscodeConfig);
+
+  onAutoFixChange(vscodeConfig);
+}
+
+function fixWorkspaceDepsIssues(vscodeConfig: VscodeFileEventConfig): void {
+  setTimeout(() => {
+    void findDepsIssuesFixThemAndCleanProblemsView(vscodeConfig);
+  }, getAutoFixDelay(vscodeConfig.workspace));
+}
+
+function getPackageJsonUris(workspace: VscodeWorkspace): Thenable<Uri[]> {
+  return workspace.findFiles("package.json", "**​/node_modules/**");
+}
+
+function onAutoFixChange(vscodeConfig: VscodeFileEventConfig): void {
+  vscodeConfig.workspace.onDidChangeConfiguration((event) => {
+    const affected = event.affectsConfiguration(ENABLE_AUTOFIX);
+    if (affected) {
+      void findDepsIssuesFixThemAndCleanProblemsView(vscodeConfig);
+    }
+  });
+}
+
+async function findDepsIssuesFixThemAndCleanProblemsView(
+  vscodeConfig: VscodeFileEventConfig
+): Promise<void> {
+  const { workspace, diagnosticCollection, outputChannel } = vscodeConfig;
+  if (!isAutoFixEnabled(workspace)) return;
+
+  const packageJsonUris = await getPackageJsonUris(workspace);
+  // TODO: should we do it in parrallel or sequentially ???
+  // currently it works sequentially
+  for (const uri of packageJsonUris) {
+    await findAndFixDepsIssues(uri, outputChannel);
+    clearDiagnostics(diagnosticCollection, uri);
```

Comment:
in regards to `clearDiagnostics` 
Why are diagnostics even involved in `auto-fix` flow?
don't they only show for opened files? and autofix works on all files in workspace?

Reply:
Not exactly. 
If a relevant package.json file is opened and has issues and auto fix runs and fixes the file, issues will still be shown in Problems view.
We need to clear them.